### ISSUE TITLE
Disconnected shaders exported

### ISSUE THREAD

**Issue by Clockwork-Muse** [KhronosGroup/glTF-Blender-IO#1329]
**Describe the bug**
Shaders that are **not** connected to the material node are still exported.  

Given a minimal node setup like this:

![image](https://user-images.githubusercontent.com/5569508/107697802-8ba24c00-6c68-11eb-81b9-b5f1b7e1943a.png)

... it still exports the emission shader data:
```json
"materials" : [
    {
        "emissiveFactor" : [
            1,
            1,
            1
        ],
        "name" : "Material",
        "pbrMetallicRoughness" : {
            "baseColorFactor" : [
                0.800000011920929,
                0.800000011920929,
                0.800000011920929,
                1
            ],
            "metallicFactor" : 0,
            "roughnessFactor" : 0.4000000059604645
        }
    }
],
```
(the reverse is also true - hook up just the emissive shader and the pbr output is still present)

This is particularly annoying when dealing with baked shader node trees, because these will be connected to an extra emission shader for the baking, and the baked result hooked up to a Principal BSDF.  In this sort of situation it's desirable to leave the original node tree in the material in case it needs to be rebaked later.

**Expected behavior**
Only the shaders currently hooked up to a material output node is exported

[Sample blend and resulting gltf](https://github.com/KhronosGroup/glTF-Blender-IO/files/5968443/untitled.zip)

**Version**
 - OS: Ubuntu 20.04 (`nvidia/opengl:1.2-glvnd-runtime-ubuntu20.04` docker image)
 - Blender Version: 2.91.2

**Comment by Clockwork-Muse** [comment]
There's an insidious twist to this:

![image](https://user-images.githubusercontent.com/5569508/107698754-e4beaf80-6c69-11eb-91bc-960173840ff7.png)
generates:
```json
"materials" : [
    {
        "name" : "Material",
        "pbrMetallicRoughness" : {
            "baseColorFactor" : [
                0.800000011920929,
                0.800000011920929,
                0.800000011920929,
                1
            ],
            "metallicFactor" : 0,
            "roughnessFactor" : 0.4000000059604645
        }
    }
],
```
... the earlier created material (the one with the default white color) is the one being exported, not the one actually hooked up to the output.  Note that if you do something like delete the earlier material and then undo the operation, the ordering changes.

**Comment by scurest** [comment]
Think this was fixed by #1249. Make sure you're testing the latest version (eg. the 2.93 alpha from builder.blender.org).

**Comment by Clockwork-Muse** [comment]
@scurest - Yes, that appears to do it.

Connecting the emissive shader results in an empty `pbrMetallicRoughness` node, is that intended?
```json
"materials" : [
    {
        "emissiveFactor" : [
            1,
            1,
            1
        ],
        "name" : "Material",
        "pbrMetallicRoughness" : {}
    }
],
```

I had initially discounted 1249 because I had assumed it had made it into Blender.  Will it get included into a 2.92 patch, or will I need to wait for 2.93 to be released?

**Comment by scurest** [comment]
> Connecting the emissive shader results in an empty pbrMetallicRoughness node, is that intended?

#1301

>  Will it get included into a 2.92 patch, or will I need to wait for 2.93 to be released?

It will be in 2.92 when it releases later this month. You can check by looking for it in the [blender-v2.92-release](https://github.com/KhronosGroup/glTF-Blender-IO/tree/blender-v2.92-release) branch.

**Comment by julienduroure** [comment]
Closing, as this is already fixed
